plugins/pi4j-plugin-ffm/src/main/java/com/pi4j/plugin/ffm/providers/spi/SpiFFM.java

```diff
@@ -87,8 +91,10 @@ public int transfer(byte[] write, int writeOffset, byte[] read, int readOffset,
         if (read != null) {
             ByteBuffer.wrap(read).put(readBytes);
             logger.trace("{} - Read buffer: {}", path, HexFormatter.format(read));
+            return readBytes.length;
+        } else {
```

Comment:
Why wrap the else block? I would think it even better if we did a read == null return ...